includes/class-wc-gateway-stripe.php

```diff
@@ -842,28 +843,37 @@ public function retry_after_error( $response, $order, $retry, $force_save_source
 	 * it up and prepare it for the Stripe PaymentIntents modal to confirm a payment.
 	 *
 	 * @since 4.2
-	 * @param WC_Payment_Gateway[] $gateways A list of all available gateways.
-	 * @return WC_Payment_Gateway[]          Either the same list or an empty one in the right conditions.
 	 */
-	public function prepare_order_pay_page( $gateways ) {
+	public function prepare_order_pay_page() {
 		if ( ! is_wc_endpoint_url( 'order-pay' ) || ! isset( $_GET['wc-stripe-confirmation'] ) ) { // wpcs: csrf ok.
-			return $gateways;
+			return;
```

Comment:
❤️ Thank you for making it clear this is an action and not a filter (here and below)

Reply:
It was a filter on the available gateways but didn't need to be AFAICT (since those are already being conditionally filtered in this function), so it's been changed to an action in this PR.